src/components/SavedRecipes.tsx

```diff
@@ -10,73 +20,123 @@ const SavedRecipes = () => {
   const [recipes, setRecipes] = useState([]);
   const [filteredRecipes, setFilteredRecipes] = useState([]);
   const [searchParams, setSearchParams] = useSearchParams();
+  const [activeCategory, setActiveCategory] = useState("");
+  const [activeTag, setActiveTag] = useState("");
   const [show, setShow] = useState(true);
   const [isLoading, setIsLoading] = useState(true);
-
   const searchQueryParam = searchParams.get("search");
+  const filteredTag = searchParams.get("filterTag") as string;
+  const filteredCategory = searchParams.get("filterCategory") as string;
```

Comment:
BLOCKING: For this and `filteredTag`, `.get()` can return a `string` or `null` so you need to handle both situations. You can't just cast directly to a string here. You'll need to handle the potential of a `null` value later on.

https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/get

Reply:
Done!